### ISSUE TITLE
 No module named 'tenacity.asyncio'

### ISSUE THREAD

**Issue by feloy** [instructlab/instructlab#1389]
**Describe the bug**

When installing ilab on Linux Fedora, I get an error `No module named 'tenacity.asyncio'`

**To Reproduce**

1. followed instruction to install ilab (tested with python 3.12 and 3.11):
```
python3 -m venv --upgrade-deps venv
source venv/bin/activate
pip cache remove llama_cpp_python
pip install instructlab --extra-index-url=https://download.pytorch.org/whl/cpu
```

2. execute ilab, I get the error:
```
$ ilab
Traceback (most recent call last):
  File "/home/phmartin/Documents/tests/InstructLab/venv/bin/ilab", line 5, in <module>
    from instructlab.lab import ilab
  File "/home/phmartin/Documents/tests/InstructLab/venv/lib64/python3.12/site-packages/instructlab/lab.py", line 18, in <module>
    from .config import config as config_group
  File "/home/phmartin/Documents/tests/InstructLab/venv/lib64/python3.12/site-packages/instructlab/config/config.py", line 8, in <module>
    from .init import init
  File "/home/phmartin/Documents/tests/InstructLab/venv/lib64/python3.12/site-packages/instructlab/config/init.py", line 13, in <module>
    from instructlab import utils
  File "/home/phmartin/Documents/tests/InstructLab/venv/lib64/python3.12/site-packages/instructlab/utils.py", line 19, in <module>
    from langchain_text_splitters import RecursiveCharacterTextSplitter
  File "/home/phmartin/Documents/tests/InstructLab/venv/lib64/python3.12/site-packages/langchain_text_splitters/__init__.py", line 22, in <module>
    from langchain_text_splitters.base import (
  File "/home/phmartin/Documents/tests/InstructLab/venv/lib64/python3.12/site-packages/langchain_text_splitters/base.py", line 23, in <module>
    from langchain_core.documents import BaseDocumentTransformer, Document
  File "/home/phmartin/Documents/tests/InstructLab/venv/lib64/python3.12/site-packages/langchain_core/documents/__init__.py", line 6, in <module>
    from langchain_core.documents.compressor import BaseDocumentCompressor
  File "/home/phmartin/Documents/tests/InstructLab/venv/lib64/python3.12/site-packages/langchain_core/documents/compressor.py", line 6, in <module>
    from langchain_core.callbacks import Callbacks
  File "/home/phmartin/Documents/tests/InstructLab/venv/lib64/python3.12/site-packages/langchain_core/callbacks/__init__.py", line 9, in <module>
    from langchain_core.callbacks.base import (
  File "/home/phmartin/Documents/tests/InstructLab/venv/lib64/python3.12/site-packages/langchain_core/callbacks/base.py", line 7, in <module>
    from tenacity import RetryCallState
  File "/home/phmartin/Documents/tests/InstructLab/venv/lib64/python3.12/site-packages/tenacity/__init__.py", line 653, in <module>
    from tenacity.asyncio import AsyncRetrying  # noqa:E402,I100
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'tenacity.asyncio'

```

**Expected behavior**

ilab should work 

**Screenshots**

**Device Info (please complete the following information):**
 - Hardware Specs: ThhinkPad p1
 - OS Version: Fedora Linux 40
 - Python Version: 3.12.2 / 3.11.9
 - InstructLab Version: ??

**Additional context**

**Comment by leseb** [comment]
The CI is also experiencing this.

**Comment by leseb** [comment]
Context https://github.com/langchain-ai/langchain/issues/22972

**Comment by leseb** [comment]
Fix https://github.com/instructlab/instructlab/issues/1389

**Comment by feloy** [comment]
> Fix #1389

You probably wanted to add this comment on https://github.com/instructlab/instructlab/pull/1390?
